static/app/views/onboarding/targetedOnboarding/components/stepper.tsx

```diff
@@ -0,0 +1,74 @@
+import styled from '@emotion/styled';
+import { animate, useMotionValue } from 'framer-motion';
+import { HTMLAttributes, useEffect, useRef } from 'react';
+import { StepDescriptor } from '../types';
+import { motion } from 'framer-motion';
+const StepperWrapper = styled('div')`
+  border-radius: 4px;
+  position: relative;
+  overflow: hidden;
+`;
+const StepperContainer = styled('div')`
+  display: flex;
+  flex-direction: row;
+  gap: 8px;
+`;
+
+const StepperIndicator = styled('span') <{ clickable?: boolean }>`
+  height: 8px;
+  width: 80px;
+  background-color: ${p => p.theme.progressBackground};
+  cursor: ${p => (p.clickable ? 'pointer' : 'default')};
+`;
+
+const StepperTransitionIndicator = styled(motion.span)`
+  height: 8px;
+  width: 80px;
+  background-color: ${p => p.theme.progressBar};
+  position: absolute;
+`;
+
+type Props = Omit<HTMLAttributes<HTMLDivElement>, 'onClick'> & {
+  steps: StepDescriptor[],
+  currentStepId: string,
+  onClick: (step: StepDescriptor) => void,
+};
+
+export default function Stepper({ steps, currentStepId, onClick, ...props }: Props) {
```

Comment:
@Neo-Zhixing should we turn this into a generic stepper component outside of the onboarding heiarchy?

Reply:
I don't current see much use for this stepper component outside the context of onboarding right now. If we need to use this component for anything else it's never too late to move a file.